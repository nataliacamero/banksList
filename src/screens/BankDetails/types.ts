import {RouteProp} from '@react-navigation/native';
import {NativeStackNavigationProp} from '@react-navigation/native-stack';
import {StackParamList} from '../../navigation/types';

export interface IResponse {
  age: number;
  bankName: string;
  description: string;
  url: string;
  id: string;
}

export type BankDetailsScreenNavigationProp = NativeStackNavigationProp<
  StackParamList,
  'BankDetails'
>;

export type BankDetailsRouteProp = RouteProp<StackParamList, 'BankDetails'>;
