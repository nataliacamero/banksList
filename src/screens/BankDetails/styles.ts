import {SafeAreaView, Text, View} from 'react-native';
import styled from 'styled-components';

export const SafeAreaStyled = styled(SafeAreaView)`
  background-color: white;
  height: 100%;
`;

export const Container = styled(View)`
  display: flex;
  flex-flow: column nowrap;
  justify-content: space-between;
  align-items: center;
  width: 100%;
  padding-top: 50px;
  padding-bottom: 50px;
`;

export const BoldParagraph = styled(Text)`
  font-size: 18px;
  font-weight: 700;
  color: #020202;
`;

export const Separator = styled(View)`
  margin-top: 20px;
`;
