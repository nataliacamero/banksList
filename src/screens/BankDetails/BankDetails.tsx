import {useRoute} from '@react-navigation/native';
import React from 'react';
import {StatusBar, Text} from 'react-native';
import {Header} from '../../components';
import {BoldParagraph, Container, SafeAreaStyled, Separator} from './styles';
import {BankDetailsRouteProp} from './types';

const BankDetails: () => JSX.Element = () => {
  const route = useRoute<BankDetailsRouteProp>();
  const {info} = route?.params;

  return (
    <SafeAreaStyled>
      <StatusBar barStyle={'dark-content'} />
      <Header
        title={'Bank Details'}
        subtitle={'Here you will find the information of the chosen Bank.'}
        testID="header-bank-details"
      />
      <Container testID={`container-sku-product`}>
        <BoldParagraph>{info.bankName}</BoldParagraph>
        <Separator />
        <Text>{info.description}.</Text>
        <Separator />
        <Text>Fecha de creacion: {info.age} a√±os.</Text>
      </Container>
    </SafeAreaStyled>
  );
};

export default BankDetails;
