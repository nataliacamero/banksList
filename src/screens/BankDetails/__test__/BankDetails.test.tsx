import React from 'react';
import {cleanup, render, screen} from '@testing-library/react-native';
import {useNavigation, useRoute} from '@react-navigation/native';
import BankDetails from '../BankDetails';

jest.useFakeTimers();
jest.mock('@react-navigation/native', () => ({
  useRoute: jest.fn(),
  useNavigation: jest.fn(),
}));

const useNavigationMock = useNavigation as jest.MockedFunction<
  typeof useNavigation
>;

afterEach(cleanup);

describe('<BankDetails/>', () => {
  beforeEach(() => {
    useNavigationMock.mockReset();
  });

  it('Should render the component', () => {
    const infoItem = {
      bankName: 'Banregio',
      description: 'Somos el banco de creadores',
      age: 10,
    };
    const route = {params: {info: infoItem}};
    (useRoute as jest.Mock).mockReturnValue(route);
    const rendered = render(<BankDetails />).toJSON();
    expect(rendered).toMatchSnapshot();
  });

  it('Should render the Title -Bank Details-', () => {
    render(<BankDetails />);
    const {getByTestId} = screen;
    const title = getByTestId('title-header-bank-details');
    expect(title.children[0]).toBe('Bank Details');
  });
  it('Should render the subtitle -Here you can...-', () => {
    render(<BankDetails />);
    const {getByTestId} = screen;
    const subtitle = getByTestId('subtitle-header-bank-details');
    expect(subtitle.children[0]).toBe(
      'Here you will find the information of the chosen Bank.',
    );
  });
});
