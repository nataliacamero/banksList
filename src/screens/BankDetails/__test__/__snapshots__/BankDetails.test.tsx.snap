// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BankDetails/> Should render the component 1`] = `
<RCTSafeAreaView
  className="sc-eDDNvR cHMPlu"
>
  <View
    className="sc-beqWaB bCUvjq"
  >
    <Text
      className="sc-gueYoa VqWBx"
      testID="title-header-bank-details"
    >
      Bank Details
    </Text>
    <Text
      className="sc-dmqHEX cobd"
      testID="subtitle-header-bank-details"
    >
      Here you will find the information of the chosen Bank.
    </Text>
  </View>
  <View
    className="sc-jTrPJq iZereL"
    testID="container-sku-product"
  >
    <Text
      className="sc-gLDzan MWLoB"
    >
      Banregio
    </Text>
    <View
      className="sc-iAEyYk khwrmI"
    />
    <Text>
      Somos el banco de creadores
      .
    </Text>
    <View
      className="sc-iAEyYk khwrmI"
    />
    <Text>
      Fecha de creacion: 
      10
       a√±os.
    </Text>
  </View>
</RCTSafeAreaView>
`;
